### ISSUE TITLE
Allow sites to define friend domains

### ISSUE THREAD

**Issue by sohomdatta1** [kyraseevers/Partitioning-visited-links-history#3]
In certain cases, multiple sites/origins could be owned by the same org (ex: `*.wikipedia.org` and `*.wikisource.org` or `*.google.com` and `*.withgoogle.com`). Could we provide some (opt-in) way/mechanism for a site to define other sites to with which `:visited` link data can be shared?

**Comment by NDevTK** [comment]
The example of "*.google.com and *.withgoogle.com" would probably be bad since I think they use separate domains for isolation.
That said sites can always "opt-in" to sharing visited links the fun way via postMessage and iframes.

**Comment by NDevTK** [comment]
At least from what I can tell *.withgoogle.com hosts less sensitive applications such as event pages.

**Comment by sohomdatta1** [comment]
> At least from what I can tell *.withgoogle.com hosts less sensitive applications such as event pages.

Yep that's a good point, whatever mechanism is implemented should probably be only one-way (`*.withgoogle.com` can share its data with `*.google.com` but unless `*.google.com` consents, `*.withgoogle.com` should not be allowed to read the visited data of `*.google.com`)

**Comment by NDevTK** [comment]
Seems like it would be better to opt-in per origin does *.google.com include insecure contexts like http://*.google.com?

**Comment by miketaylr** [comment]
@sohomdatta1 could you expand on the use case a bit more? I'm having a hard time understanding how this would be useful. Thanks!
